File: app/src/main/java/me/tatarka/betterfragment/sample/dagger/AppComponent.java
package me.tatarka.betterfragment.sample.dagger;

import javax.inject.Singleton;

import dagger.Component;
import me.tatarka.betterfragment.sample.MainActivity;
import me.tatarka.betterfragment.sample.SimpleHostFragment;

@Singleton
@Component(modules = FragmentModule.class)
public interface AppComponent {
    AppComponent INSTANCE = DaggerAppComponent.create();

    void inject(SimpleHostFragment fragment);

    void inject(DaggerFragmentHost host);

    void inject(DaggerFragmentNavHost navHost);

    void inject(MainActivity activity);
}

File: app/src/main/java/me/tatarka/betterfragment/sample/dagger/AppComponent.kt
package me.tatarka.betterfragment.sample.dagger

import javax.inject.Singleton

import dagger.Component
import me.tatarka.betterfragment.sample.MainActivity
import me.tatarka.betterfragment.sample.SimpleHostFragment

@Singleton
@Component(modules = [FragmentModule::class])
interface AppComponent {
    val fragmentFactory: DaggerFragmentFactory
}

val injector: AppComponent = DaggerAppComponent.create()
